common/app/templates/inlineJS/nonBlocking/detectAdblock.scala.js

```diff
@@ -0,0 +1,24 @@
+try {
+    ((document, window) => {
+        var ad = document.createElement('div');
+        ad.style.position = 'absolute';
+        ad.style.left = '-9999px';
+        ad.style.height = '10px';
+        ad.innerHTML = '&nbsp;';
+        ad.setAttribute('class', 'ad_unit');
+
+        window.requestAnimationFrame(() => {
+            document.body.appendChild(ad);
+
+            window.requestAnimationFrame(() => {
```

Comment:
@regiskuckaertz We did test whether we could use mutation observers to intercept ABP changes, at least when trying to improve FF ABP false-positives, and the answer was: no. This might be a design decision not to allow web pages to sniff information about running plugins.

We spent several weeks investigating Adblock detection; I can show you the data we collected, if you like. Long story short - it's complicated, different plugins have different implementations, and the only real commonality you can rely upon is that they all reject connections to URIs matching particular text patterns.